task2/app.py

```diff
@@ -3,32 +3,52 @@
 # 
 
 class PhoneBook:
+    
     def __init__(self) -> None:
-        pass
+        self.contacts = []
 
     def add(self, name: str, surname: str, phone: str) -> None:
-        """??? (https://peps.python.org/pep-0257/)"""
-        pass
+        """Adds a person to the phone book."""
+        for contact in self.contacts:
+            if contact['name'] == name and contact['surname'] == surname:
+                return
+        self.contacts.append({'name': name, 'surname': surname, 'phone': phone})
 
     def update(self, name: str, surname: str, phone: str) -> bool:
-        """??? (https://peps.python.org/pep-0257/)"""
-        pass
+        """Updates information about a person in the phone book."""
+        for contact in self.contacts:
+            if contact['name'] == name and contact['surname'] == surname:
+                contact['phone'] = phone
 
     def remove(self, name: str, surname: str) -> int:
-        """??? (https://peps.python.org/pep-0257/)"""
-        pass
+        """Removes a person from the phone book."""
+        for contact in self.contacts:
+            if contact['name'] == name and contact['surname'] == surname:
+                self.contacts.remove(contact)
+            return 1
+        return 0
 
     def search(self, name: str = None, surname: str = None) -> list[dict]:
-        """??? (https://peps.python.org/pep-0257/)"""
-        return []
+        """Searches for a person in the phone book."""
+        if name == None and surname == None:
+            return self.contacts
```

Comment:
Please, use `is` here instead of `==`

See (second dot): https://peps.python.org/pep-0008/#programming-recommendations

> Comparisons to singletons like None should always be done with is or is not, never the equality operators.